easybuild/easyblocks/x/xalt.py

```diff
@@ -0,0 +1,92 @@
+##
+# Copyright 2020 NVIDIA
+#
+# This file is triple-licensed under GPLv2 (see below), MIT, and
+# BSD three-clause licenses.
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for XALT, implemented as an easyblock
+@author: Scott McMillan (NVIDIA)
+"""
+import os
+import re
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM, MANDATORY
+from easybuild.tools.build_log import EasyBuildError
+
+
+class EB_XALT(ConfigureMake):
+    """Support for building and installing XALT."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'config_py': [None, "XALT site filter file", MANDATORY],
+            'syshost': ['hardcode:cluster', "System name", CUSTOM],
+            'transmission': ['syslog', "", CUSTOM],
+        }
```

Comment:
I think it should have these additional extra options
```
'executable_tracking': ['yes', '', CUSTOM],
'scalar_sampling': ['yes', '', CUSTOM],
'linker': ['ld', '', CUSTOM],
'logging_url': [None, 'logging url when using curl transmission method', CUSTOM],
```

Reply:
I agree that having easyconfig parameters for `executable_tracking` and `scalar_sampling` makes sense, and having them enabled by default makes sense too. That way the corresponding environment variables and/or configuration options (probably only the latter?) can be put in place by the easyblock...

Same goes for GPU tracking I guess, since having a `gpu_tracking` option with default `None` could maybe even allow EasyBuild to figure out whether or not GPU tracking *can* be enabled (based on whether the required dependencies are available, like `nvidia.h` and CUDA).